packages/ringcentral-widgets/containers/CallCtrlPage/index.js

```diff
@@ -125,46 +140,131 @@ class CallCtrlPage extends Component {
+            this.setState(prev => ({
+              ...prev,
+              lastTo: {
+                calleeType: calleeTypes.unknow,
+                avatarUrl: null,
+                sessionId: lastCall.webphoneSession ? lastCall.webphoneSession.id : null,
+                status: lastCall.webphoneSession ? lastCall.webphoneSession.callStatus : null,
+                name: lastCall.to.phoneNumber
+              }
+            }));
+          }
         }
       }
     } else {
-      const { conferencePartiesAvatarUrls } = this.props;
-      this.setState({
+      const { conferencePartiesAvatarUrls, conferenceCall } = this.props;
+      if (Object.values(conferenceCall.conferences).length) {
+        this.lastToListen(Object.values(conferenceCall.conferences)[0].session.id);
+        this.setState(() => ({
+          lastTo: {
+            calleeType: calleeTypes.conference,
+            avatarUrl: conferencePartiesAvatarUrls[0],
+            extraNum: conferencePartiesAvatarUrls.length - 1,
+            sessionId: Object.values(conferenceCall.conferences)[0].session.id
+          }
+        }));
+      }
+    }
+  }
+  updateLastToState(nextProps) {
+    if (Object.keys(nextProps.conferenceCall.conferences).length) {
+      const part = this.props.getPartyProfiles().length - 1;
+      if (!this._mounted) {
+        return;
+      } else {
+        this.setState(prev => ({
+          lastTo: {
+            ...prev.lastTo,
+            extraNum: part
+          }
+        }));
+      }
+    }
+  }
+  terminatedCallback(session, isLastTo = false) {
+    return () => this.terminatedSuccessHander(session, isLastTo);
+  }
+  terminatedSuccessHander(session, isLastTo) {
+    session.removeListener('terminated', this.terminatedSuccessHander);
+    const { routerInteraction, webphone, conferenceCall } = this.props;
+    if (isLastTo) {
+      this.setState(prev => ({
         lastTo: {
-          calleeType: calleeTypes.conference,
-          avatarUrl: conferencePartiesAvatarUrls[0],
-          extraNum: conferencePartiesAvatarUrls.length - 1,
+          ...prev.lastTo,
+          status: sessionStatus.finished
         },
-      });
+        mergeDisabled: true
+      }));
+      if (!conferenceCall.isConferenceSession(webphone.activeSessionId)) {
+        if (webphone.activeSession) {
+          setTimeout(() => {
+            routerInteraction.push('calls/active');
+          }, 2000);
+        } else {
+          routerInteraction.push('/dialer');
+        }
+      }
+    } else if (!conferenceCall.isConferenceSession(webphone.activeSessionId)) {
+      if (webphone.activeSession) {
+        routerInteraction.push('calls/active');
+      } else {
+        routerInteraction.push('/dialer');
+      }
+    }
+  }
+  lastToListen(sessionId) {
+    const { webphone } = this.props;
+    const session = webphone._sessions.get(sessionId);
+    session.on('terminated', this.terminatedCallback(session, true));
```

Comment:
and again, you hit the GC issue, please give the handler a specific reference so the V8 can do GC on it